src/lib/pubkey/ec_group/ec_inner_pc.cpp

```diff
@@ -0,0 +1,209 @@
+      m_group(std::move(group)),
+      m_pt(deserialize_pcurve_pt(m_group->pcurve(), bytes)),
+      m_bytes(m_pt.serialize<secure_vector<uint8_t>>()),
+      m_x_bytes(std::span{m_bytes}.subspan(1, field_element_bytes())),
+      m_y_bytes(std::span{m_bytes}.last(field_element_bytes())) {}
+
+const EC_AffinePoint_Data_PC& EC_AffinePoint_Data_PC::checked_ref(const EC_AffinePoint_Data& data) {
+   const auto* p = dynamic_cast<const EC_AffinePoint_Data_PC*>(&data);
+   if(!p) {
+      throw Invalid_State("Failed conversion to EC_AffinePoint_Data_PC");
+   }
+   return *p;
+}
+
+std::unique_ptr<EC_AffinePoint_Data> EC_AffinePoint_Data_PC::clone() const {
+   return std::make_unique<EC_AffinePoint_Data_PC>(m_group, m_pt);
+}
+
+const std::shared_ptr<const EC_Group_Data>& EC_AffinePoint_Data_PC::group() const {
+   return m_group;
+}
+
+std::unique_ptr<EC_AffinePoint_Data> EC_AffinePoint_Data_PC::mul(const EC_Scalar_Data& scalar,
+                                                                 RandomNumberGenerator& rng,
+                                                                 std::vector<BigInt>& ws) const {
+   BOTAN_UNUSED(ws);
+
+   BOTAN_ARG_CHECK(scalar.group() == m_group, "Curve mismatch");
+   const auto& k = EC_Scalar_Data_PC::checked_ref(scalar).value();
+   auto pt = m_group->pcurve().mul(m_pt, k, rng).to_affine();
+   return std::make_unique<EC_AffinePoint_Data_PC>(m_group, std::move(pt));
+}
+
+size_t EC_AffinePoint_Data_PC::field_element_bytes() const {
+   return m_group->pcurve().field_element_bytes();
+}
+
+void EC_AffinePoint_Data_PC::serialize_x_to(std::span<uint8_t> bytes) const {
+   const size_t fe_bytes = this->field_element_bytes();
+   BOTAN_ARG_CHECK(bytes.size() == fe_bytes, "Invalid output size");
+   copy_mem(bytes, m_x_bytes);
+}
+
+void EC_AffinePoint_Data_PC::serialize_y_to(std::span<uint8_t> bytes) const {
+   const size_t fe_bytes = this->field_element_bytes();
+   BOTAN_ARG_CHECK(bytes.size() == fe_bytes, "Invalid output size");
+   copy_mem(bytes, m_y_bytes);
+}
+
+void EC_AffinePoint_Data_PC::serialize_xy_to(std::span<uint8_t> bytes) const {
+   const size_t fe_bytes = this->field_element_bytes();
+   BOTAN_ARG_CHECK(bytes.size() == 2 * fe_bytes, "Invalid output size");
+   copy_mem(bytes, std::span{m_bytes}.last(2 * fe_bytes));
+}
+
+void EC_AffinePoint_Data_PC::serialize_compressed_to(std::span<uint8_t> bytes) const {
+   const size_t fe_bytes = this->field_element_bytes();
+   BOTAN_ARG_CHECK(bytes.size() == 1 + fe_bytes, "Invalid output size");
+   const bool y_is_odd = (m_bytes.back() & 0x01) == 0x01;
+
+   BufferStuffer stuffer(bytes);
+   stuffer.append(y_is_odd ? 0x03 : 0x02);
+   this->serialize_x_to(stuffer.next(fe_bytes));
+}
+
+EC_Point EC_AffinePoint_Data_PC::to_legacy_point() const {
+   auto x = BigInt::from_bytes(m_x_bytes);
+   auto y = BigInt::from_bytes(m_y_bytes);
+   return EC_Point(m_group->curve(), std::move(x), std::move(y));
```

Comment:
Nit: The modern C++ language nerds consider explicit `std::move` invocations as a code smell (outside of c'tor member initializer lists). Technically, this could be:

```suggestion
   return EC_Point(m_group->curve(), BigInt::from_bytes(m_x_bytes), BigInt::from_bytes(m_y_bytes));
```

... without a real impact on readability in this particular instance.

Without saying that it is relevant for performance here, but passing the rvalue directly into the function is also more efficient. The compiler can just create the return value of `from_bytes()` right into the argument location of the `EC_Point()` constructor. There won't be an invocation of the move c'tor of `BigInt` and the short-lived (moved-from) `BigInt` object also doesn't need to be destructed.

See: https://godbolt.org/z/c1TK3v9nK